### ISSUE TITLE
basic_parser doesn't work correctly when expression has single capture group

### ISSUE THREAD

**Issue by cheezman34** [andychase/reparse#1]
With a single capture group in an expression, basic_parser only returns the first digit of the text it matches.  basic_parser works correctly with multiple capture groups in an expression, and alt_parser works correctly regardless.  Example:

```
import reparse

expressions = {
    'minute':{
        'yolo':{
            'Expression': '(\d{2})',
            'Groups'    : ['mygroup']
        }
    }
}

patterns = {
    'minute_pattern':{
        'Pattern' : '<minute>'
    }
}

def process_yolo(**kwargs):
    print str(kwargs)

functions = {
    'yolo' : process_yolo,
}

minute_parser = reparse.parser(functions=functions, expressions=expressions, patterns=patterns)
minute_parser('hey buddy its 30 now')
#prints: {'mygroup': '3'} --> only gets the first digit when you'd expect both

minute_parser = reparse.parser(reparse.parsers.alt_parser, functions=functions, 
                            expressions=expressions, patterns=patterns)
minute_parser('hey buddy its 30 now')
#prints {'mygroup': '30'} --> alt_parser correctly gets both digits
```

I think the issue stems from how regex returns results from findall - with a single capture group, findall returns a list with one element, but with multiple capture groups, it returns a list with a tuple inside, like so:

```
>>> import regex
>>> re.findall('(\d+)', 'hey buddy its 30 now')
['30']
>>> re.findall('(\d+)()', 'hey buddy its 30 now')
[('30', '')]
```

Because alt_parser uses regex.scanner, it sidesteps the issue.

**Comment by andychase** [comment]
Thanks for reporting this cheezman!

Ahh Python regex creating little corner cases, how tricky. I'll look into making a patch for this shortly.

**Comment by andychase** [comment]
Fixed in version 2.1 (47ee095c6226663d92728cf107f9a48ff49fd550).

The workaround I chose was simply checking if the `findall` result was a string and if so just wrap it. See change in ([Fix only one capture group causing problems](https://github.com/andychase/reparse/commit/1389e2a60d339764516bcef4d5125d8d75df8af3#diff-25c722ae3a27d65fb38671a9a3a327dbR37)).

Thanks again @cheezman34 for reporting this!
